contentcuration/contentcuration/viewsets/community_library_submission.py

```diff
@@ -0,0 +1,145 @@
+        list_serializer_class = BulkListSerializer
+
+    def create(self, validated_data):
+        channel = validated_data["channel"]
+        user = self.context["request"].user
+
+        if channel.version == 0:
+            # The channel is not published
+            raise ValidationError(
+                "Cannot create a community library submission for an "
+                "unpublished channel."
+            )
+
+        if not channel.editors.filter(id=user.id).exists():
+            raise ValidationError(
+                "Only editors can create a community library "
+                "submission for this channel."
+            )
+
+        validated_data["channel_version"] = channel.version
+        validated_data["author"] = self.context["request"].user
+
+        countries = validated_data.pop("countries", [])
+        instance = super().create(validated_data)
+
+        for country in countries:
+            instance.countries.add(country)
+
+        instance.save()
+        return instance
+
+    def update(self, instance, validated_data):
+        if instance.channel.id != validated_data["channel"].id:
+            raise ValidationError(
+                "Cannot change the channel corresponding to "
+                "a community library submission."
+            )
+
+        countries = validated_data.pop("countries", [])
+        instance.countries.set(countries)
+
+        return super().update(instance, validated_data)
+
+
+class CommunityLibrarySubmissionPagination(ValuesViewsetCursorPagination):
+    ordering = "-date_created"
+    page_size_query_param = "max_results"
+    max_page_size = 100
+
+
+class CommunityLibrarySubmissionViewSet(
+    RESTCreateModelMixin,
+    RESTUpdateModelMixin,
+    RESTDestroyModelMixin,
+    ReadOnlyValuesViewset,
+):
+    values = (
+        "id",
+        "description",
+        "channel_id",
+        "channel_version",
+        "author_id",
+        "author__first_name",
+        "author__last_name",
+        "categories",
+        "date_created",
+        "status",
+    )
+    field_map = {
+        "author_first_name": "author__first_name",
+        "author_last_name": "author__last_name",
+    }
+    filter_backends = [DjangoFilterBackend]
+    filterset_fields = ["channel"]
+    permission_classes = [IsAuthenticated]
+    pagination_class = CommunityLibrarySubmissionPagination
+    serializer_class = CommunityLibrarySubmissionSerializer
+
+    def get_queryset(self):
+        base_queryset = CommunityLibrarySubmission.objects.all()
+        user = self.request.user
+        queryset = CommunityLibrarySubmission.filter_view_queryset(
+            base_queryset, user
+        ).order_by("-date_created")
+
+        return queryset
+
+    def get_edit_queryset(self):
+        base_queryset = CommunityLibrarySubmission.objects.all()
+        user = self.request.user
+        queryset = CommunityLibrarySubmission.filter_edit_queryset(base_queryset, user)
+
+        return queryset
```

Comment:
Oh, we don't need to override these methods, we already have them thanks to our BaseValuesViewset from which the ReadOnlyValuesViewset inherits: https://github.com/jakoma02/studio/blob/9e7c0f7fff6f749bc04b8acb8635f848eaa5823d/contentcuration/contentcuration/viewsets/base.py#L581-L595.

So we can define our base queryset directly as a class property. (e.g. https://github.com/jakoma02/studio/blob/9e7c0f7fff6f749bc04b8acb8635f848eaa5823d/contentcuration/contentcuration/viewsets/assessmentitem.py#L247) and we can apply the order_by there.